### DISCUSSION TITLE
how to create loop in RDL where the LIMIT is set regfile parameter

### DISCUSSION THREAD

**Discussion by assadk-g** [SystemRDL/systemrdl-compiler#203] [Q&A] [answered]
tried the below code .. seems it is not working 

```
regfile block_regfile#(
       longint unsigned LIMIT = 10
){

  <% for( $i = 0 ; $i < **LIMIT**; $i += 1 ) { %> 
      reg {
               field { desc = "csr_register: <%=$i%> ";
                sw = rw; hw = rw; } value[31:0] = 32'h0;
             } csr_register<%=$i%>;
 <% } %>

};
```

any ideas - how such cases can be addressed ?

**Comment by amykyta3** [comment, marked as answer]
By definition, the Perl preprocessor is a text-transformation step that happens before RDL lexing and parsing, so parameters are not accessible to it in the way you are attempting. Parameters are elaborated at the last step after compilation during elaboration - well after the preprocessor.

Why not use an array instead?

```
regfile block_regfile#(
    longint unsigned LIMIT = 10
){
    reg {
        field {
            sw = rw; hw = rw; } value[31:0] = 32'h0;
        } csr_register[LIMIT];
    };
};
```

**Comment by assadk-g** [comment]
indeed..i'll try to apply it as an array .. 
Thanks
